File: app/src/main/java/com/example/user/fuckinggreatadvicegetter/MainActivity.java
package com.example.user.fuckinggreatadvicegetter;

import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import io.realm.Realm;


public class MainActivity extends AppCompatActivity {

    ViewPager viewPager;
    TabLayout tabLayout;
    AdviceAdapter adapter;
    Realm realm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        realm.init(this);
        viewPager = (ViewPager) findViewById(R.id.mainActivityPager);
        tabLayout = (TabLayout) findViewById(R.id.tablayout);
        setupViewPager(viewPager);
        tabLayout.setupWithViewPager(viewPager);
    }

    private void setupViewPager(ViewPager viewPager) {
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new AdviceFragment(), getResources().getString(R.string.advice));
        adapter.addFragment(new FavoritesFragment(), getResources().getString(R.string.favorites));
        viewPager.setAdapter(adapter);
    }

    public void setAdapter(AdviceAdapter adapter) {
        this.adapter = adapter;
    }

    public AdviceAdapter getAdapter() {
        return adapter;
    }
}

File: app/src/main/java/com/example/user/fuckinggreatadvicegetter/views/MainActivity.kt
package com.example.user.fuckinggreatadvicegetter.views

import android.os.Bundle
import android.support.design.widget.BottomNavigationView.OnNavigationItemSelectedListener
import android.support.v7.app.AppCompatActivity
import com.example.user.fuckinggreatadvicegetter.R
import com.example.user.fuckinggreatadvicegetter.views.adapters.FavoritesListAdapter
import io.realm.Realm
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var adapter: FavoritesListAdapter? = null
    private val adviceFragment = AdviceFragment()
    private val favoritesFragment = FavoritesFragment()
    private val fragmentManager = supportFragmentManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Realm.init(this)
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        fragmentManager.beginTransaction().add(R.id.frame_container, adviceFragment).commit()
    }

    fun setAdapter(adapter: FavoritesListAdapter) {
        this.adapter = adapter
    }

    fun getAdapter(): FavoritesListAdapter? {
        return adapter
    }

    private val mOnNavigationItemSelectedListener = OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.adviceFragment -> {
                fragmentManager.beginTransaction().replace(R.id.frame_container, adviceFragment).commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.favoritesFragment -> {
                fragmentManager.beginTransaction().replace(R.id.frame_container, favoritesFragment).commit()
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }
}